src/fides/api/db/crud.py

```diff
@@ -324,6 +327,19 @@ async def delete_resource(
                     log.debug("Deleting resource")
                     query = _delete(sql_model).where(sql_model.fides_key == fides_key)
                 await async_session.execute(query)
+            except IntegrityError as err:
+                raw_error_text: str = err.orig.args[0]
+
+                if "violates foreign key constraint" in raw_error_text:
+                    error_message = "Failed to delete resource! Foreign key constraint found, try deleting related objects first."
```

Comment:
@NevilleS we can play with the wording here but generally I thought this was clear

Reply:
i know you didn't ask me, but it looks pretty clear to me :) my only nit may be to change "object" to "resource"? but i'm not really sure, maybe that's less clear!
```suggestion
                    error_message = "Failed to delete resource! Foreign key constraint found, try deleting related resources first."
```